_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-instagram-gallery.php

```diff
@@ -184,8 +200,26 @@ public function delete_instagram_access_token( $request ) {
 	 * @return mixed
 	 */
 	public function get_instagram_gallery( $request ) {
+		if ( $this->is_wpcom() ) {
+			return WPCOM_Instagram_Gallery_Helper::get_gallery( $request['access_token'], $request['count'] );
+		}
+
 		return Jetpack_Instagram_Gallery_Helper::get_instagram_gallery( $request['access_token'], $request['count'] );
 	}
+
+	/**
+	 * Determine if we are on WordPress.com,
+	 * and conditionally requires the WPCOM Instagram Gallery Helper library.
+	 *
+	 * @return boolean
+	 */
+	private function is_wpcom() {
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			require_lib( 'instagram-gallery-helper' );
```

Comment:
This looks like a side effect because of the name of the function. I wonder if we can pick a different name, maybe `handle_wpcom`?

Reply:
I agree that it definitely smells.
I kinda went for this anyway because it saved a line per each function.

I guess another way could be to have `$is_wpcom` as a class property, set it in the constructor, and require the lib in the constructor as well.

Reply:
Oh yeah that's a nice idea